### DISCUSSION TITLE
ScriptExternalLoadError: Loading script failed

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [webpack/webpack#13271] [Q&A] [answered]
I'm trying to share a hook between two applications that are in different repositories. I catch error

```js
// yes, i install babel plugin and set webpack flag 
const useUserLogin = await import("CorpCRM/userUserLogin")


```

```js
// hub app
 new ModuleFederationPlugin({
            name: 'CorpCRM',
            filename: 'remoteEntry.js',
            exposes: {
                useUserLogin: './src/hooks/useUserLogin',
            },
            remotes: {
                app2: 'app2@http://localhost:3002/remoteEntry.js',
            },
            shared: [{ react: { singleton: true }, 'react-dom': { singleton: true } }],
        })

```

```js
// second app
    new ModuleFederationPlugin({
      name: "app2",
      filename: "remoteEntry.js",
      remotes: {
        CorpCRM: 'CorpCRM@http://localhost:7777/remoteEntry.js',
      },
      shared: [{ react: { singleton: true }, 'react-dom': { singleton: true }}],
    })
```

![image](https://user-images.githubusercontent.com/19804652/116566491-7c4da900-a90f-11eb-9e77-bba0e54f1011.png)
![image](https://user-images.githubusercontent.com/19804652/116567122-039b1c80-a910-11eb-85f7-4e9b09efc957.png)

**Comment by alexander-akait** [comment, marked as answer]
Try to open `http://localhost:7777/remoteEntry.js`, looks you have wrong URL

**Reply by popuguytheparrot** [reply]
Last screenshot

**Reply by alexander-akait** [reply]
How you import `./src/hooks/useUserLogin.ts`?

**Reply by popuguytheparrot** [reply]
const useUserLogin = await import("CorpCRM/userUserLogin")

yes, i install babel plugin and set webpack flag

**Reply by alexander-akait** [reply]
hm, strange, maybe you can create simple example (just remove most of code) of the problem?

**Reply by popuguytheparrot** [reply]
Left to do

**Reply by popuguytheparrot** [reply]
@alexander-akait https://github.com/popuguytheparrot/module-federation-examples/tree/master/reproduce-repo

**Reply by popuguytheparrot** [reply]
any updates?

**Reply by alexander-akait** [reply]
Be patient, I'll see soon

**Reply by alexander-akait** [reply]
Maybe you have browser cache? because when I change (**app2/webpack.config.js**):
```js
new ModuleFederationPlugin({
      name: "app2",
      filename: "remoteEntry.js",
      remotes: {
        // No 7777 port
        CorpCRM: 'CorpCRM@http://localhost:3001/remoteEntry.js',
      },
      shared: [{ react: { singleton: true }, 'react-dom': { singleton: true }}],
}),
```

And change (**app2/src/App.tsx**):
```js
const useUserLogin = await import("CorpCRM/useUserLogin")
```

to

```js
const useUserLogin = (await import("CorpCRM/useUserLogin")).default
```

all works fine
![Screenshot from 2021-04-30 17-44-15](https://user-images.githubusercontent.com/4567934/116714296-4a574800-a9de-11eb-8544-7a89ca810305.png)

Note: you need run both server, i.e.
```
cd reproduce-repo
// In one TTY
cd app1 && npm install && npm run start
// In second TTY
cd app2 && npm run start
```

**Reply by popuguytheparrot** [reply]
Now catch this error
![image](https://user-images.githubusercontent.com/19804652/116781464-b8a61400-aa8b-11eb-9077-bd011be0cfb8.png)

**Reply by alexander-akait** [reply]
You have 404, it means you still have invalid port/host/etc, double check it, when you run `cd app1 && npm install && npm run start` you should see in console `3001` port, when you run `cd app2 && npm run start` you should see in console `3002`, and please open **two** console windows, and run **two** dev server, for first you need to run `cd app1 && npm install && npm run start`, for second you need to run `cd app2 && npm install && npm run start`

**Comment by knightswatch3** [comment]
I have a similar issue. There are two apps (app1 and app2) and everytime I import app1 which is a federated module and try to use methods of the exported content from remote, the value is undefined. And this is leading to the ScriptLoadErrors.
All my endpionts are valid as I verified they are loaded in the UI.
